lib/types/spec.ex

```diff
@@ -2,11 +2,15 @@ defmodule Soroban.Types.Spec do
   @moduledoc """
   Defines base types constructions.
   """
-  alias Stellar.TxBuild.SCVal
+  alias Stellar.TxBuild.{SCMapEntry, SCVal}
 
   @type error :: {:error, atom()}
   @type sc_val :: SCVal.t()
+  @type map_entry :: SCMapEntry.t()
 
   @callback new(any()) :: struct() | error()
   @callback to_sc_val(struct()) :: sc_val()
+  @callback to_map_entry(struct()) :: map_entry()
```

Comment:
IMO it's neccessary to add the `sc` to avoid confussions between `Soroban.Types.MapEntry` and `Stellar.TxBuild.SCMapEntry`. Also adding `sc` follows the same convention as `to_sc_val`
```suggestion
  @callback to_sc_map_entry(struct()) :: sc_map_entry()
```